client/my-sites/checkout/cart/cart-free-user-plan-upsell.jsx

```diff
@@ -0,0 +1,164 @@
+import { isDomainRegistration } from 'lib/products-values';
+
+class CartFreeUserPlanUpsell extends React.Component {
+	static propTypes = {
+		cart: PropTypes.object,
+		selectedSite: PropTypes.oneOfType( [ PropTypes.object, PropTypes.bool ] ),
+		hasPaidPlan: PropTypes.bool,
+		hasPlanInCart: PropTypes.bool,
+		isPlansListFetching: PropTypes.bool,
+		isRegisteringDomain: PropTypes.bool,
+		isSitePlansListFetching: PropTypes.bool,
+		personalPlan: PropTypes.oneOfType( [ PropTypes.object, PropTypes.bool ] ),
+		planPrice: PropTypes.oneOfType( [ PropTypes.number, PropTypes.bool ] ),
+		showPlanUpsell: PropTypes.bool,
+		translate: PropTypes.func.isRequired,
+	};
+
+	isLoading() {
+		const { cart } = this.props;
+		const isLoadingCart = ! cart.hasLoadedFromServer || cart.hasPendingServerUpdates;
+		const isLoadingPlans = this.props.isPlansListFetching;
+		const isLoadingSitePlans = this.props.isSitePlansListFetching;
+		return isLoadingCart || isLoadingPlans || isLoadingSitePlans;
+	}
+
+	getUpgradeText() {
+		const { cart, planPrice, translate } = this.props;
+		const firstDomain = find( getAllCartItems( cart ), isDomainRegistration );
+
+		if ( planPrice > firstDomain.cost ) {
+			const extraToPay = planPrice - firstDomain.cost;
+			return translate(
+				'Pay an {{strong}}extra %(extraToPay)s{{/strong}} for our Personal plan, and get access to all its ' +
+					'features, plus the first year of your domain for free.',
+				{
+					args: {
+						extraToPay: formatCurrency( extraToPay, firstDomain.currency ),
+					},
+					components: {
+						strong: <strong />,
+					},
+				}
+			);
+		} else if ( planPrice < firstDomain.cost ) {
+			const savings = firstDomain.cost - planPrice;
+			return translate(
+				'{{strong}}Save %(savings)s{{/strong}} when you purchase a WordPress.com Personal plan ' +
+					'instead — your domain comes free for a year.',
+				{
+					args: {
+						savings: formatCurrency( savings, firstDomain.currency ),
+					},
+					components: {
+						strong: <strong />,
+					},
+				}
+			);
+		}
+
+		return translate(
+			'Purchase our Personal plan at {{strong}}no extra cost{{/strong}}, and get access to all its ' +
+				'features, plus the first year of your domain for free.',
+			{
+				components: {
+					strong: <strong />,
+				},
+			}
+		);
+	}
+
+	shouldRender() {
+		if ( this.isLoading() ) {
+			return false;
+		}
+
+		const {
+			hasPaidPlan,
+			hasPlanInCart,
+			isRegisteringDomain,
+			selectedSite,
+			showPlanUpsell,
+		} = this.props;
+
+		return (
+			isRegisteringDomain && showPlanUpsell && selectedSite && ! hasPaidPlan && ! hasPlanInCart
+		);
+	}
+
+	addPlanToCart() {
+		const planCartItem = planItem( PLAN_PERSONAL, {} );
+		addItem( planCartItem );
+	}
+
+	render() {
+		if ( ! this.shouldRender() ) {
+			return null;
+		}
+
+		const { translate } = this.props;
+
+		return (
+			<div>
+				<SectionHeader className="cart__header" label={ translate( 'Upgrade and save' ) } />
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'Upgrade and Save!' )` *ES Score: 8*